samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <shell/shell.h>
+
+#include <getopt.h>
+
+#include <modem/location.h>
+
+#include "mosh_print.h"
+
+enum location_shell_command {
+	LOCATION_CMD_NONE = 0,
+	LOCATION_CMD_GET,
+	LOCATION_CMD_CANCEL
+};
+
+struct location_shell_cmd_args {
+	enum location_shell_command command;
+};
+
+/******************************************************************************/
+static const char location_usage_str[] =
+	"Usage: location <subcommand> [options]\n"
+	"\n"
+	"Subcommands:\n"
+	"  get:      Requests the current position or starts periodic position updates.\n"
+	"  cancel:   Cancel/Stop on going request. No options\n";
+
+static const char location_get_usage_str[] =
+	"Usage: location get [--method <method>] [--interval <secs>]\n"
+	"[--gnss_accuracy <acc>] [--gnss_timeout <timeout in secs>]\n"
+	"[--cellular_timeout <timeout in secs>] [--cellular_service <service_string>]\n"
+	"[--wifi_timeout <timeout in secs>] [--wifi_service <service_string>]\n"
+	"Options:\n"
+	"  --method,           Location method: 'gnss', 'cellular' or 'wifi'. Multiple\n"
+	"                      '--method' parameters may be given to indicate list of\n"
+	"                      methods in priority order.\n"
+	"  --interval,         Position update interval in seconds\n"
+	"                      (default: 0 = single position)\n"
+	"  --gnss_accuracy,    Used GNSS accuracy: 'low' or 'normal'\n"
```

Comment:
At least for GNSS accuracy the default could be mentioned, that's not going to change. I'm not sure whether we should also mention default timeouts or not?

Reply:
Decided not to add defaults